### PR TITLE
 Switch amber to parse clspv's embedded reflection

### PR DESCRIPTION
**Pull request by alan-baker**
* No longer uses the descriptor map, instead parses the embedded
  reflection instructions directly
* strips reflection instructions after parsing to prevent needing
  VK_KHR_non_semantic_info

### CODE REVIEW

File: src/clspv_helper.cc [hunk 1]
```diff
@@ -14,117 +14,267 @@
...
+#include "spirv/unified1/spirv.hpp"
+#include "spirv-tools/libspirv.hpp"
+#include "spirv-tools/optimizer.hpp"
+
+using amber::Pipeline;
+
+namespace {
+
+struct ReflectionHelper {
+  Pipeline::ShaderInfo* shader_info;
```
Comment (dj2) [inline src/clspv_helper.cc:33]: Should these two default to nullptr?

File: src/clspv_helper.cc [hunk 2]
```diff
@@ -14,117 +14,267 @@
...
+      break;
+    } break;
+  }
+
+  return SPV_SUCCESS;
+}
+
+spv_result_t ParseReflection(void* user_data,
+                             const spv_parsed_instruction_t* inst) {
+  ReflectionHelper* helper = reinterpret_cast<ReflectionHelper*>(user_data);
```
Comment (dj2) [inline src/clspv_helper.cc:206]: nit: this could be `auto*`

File: src/clspv_helper.h [hunk 3]
```diff
@@ -18,6 +18,8 @@
 #include <string>
 #include <vector>
 
+#include "spirv-tools/libspirv.h"
+
```
Comment (dj2) [inline src/clspv_helper.h:22]: nit: no blank line

File: src/shader_compiler.h [hunk 4]
```diff
@@ -19,6 +19,8 @@
 #include <utility>
 #include <vector>
 
+#include "spirv-tools/libspirv.h"
+
```
Comment (dj2) [inline src/shader_compiler.h:23]: ditto

**Comment by alan-baker** [comment]
https://github.com/microsoft/DirectXShaderCompiler/pull/3074 is necessary to get DXC compatible with the updated SPIRV-Tools

**Pull request merged**